pitch.md

```diff
@@ -23,38 +86,122 @@ should be more specific than "The user describes properties of unicorns.", e.g.,
 by describing which unicorn properties (horn color, tattoo, etc.) are most 
 important to users and what kinds of values those properties might have._ 
 
+The user creates instruments by combining components. There are three types of
+components, distinguished by their inputs and outputs.
+
+A _source_ is a component which produces a signal. (A signal is like a
+potential sound. At runtime, it can be synthesized into a sound via the
+computer's DAC, or perhaps it can be written to a file.) A source has no input
+(though it may have _parameters_, see below) and a single output. There will be
+several types of primitive sources, distinguished by the characteristics of the
+signals they produce. Examples include frequency modulation synthesizers and
+physical modeling synthesizers.
+
+An _effect_ is a component which modifies a signal. Its input is a signal, and
+its output is a new, transformed signal. Examples of primitive effects may
+include equalization, compression, and distortion.
```

Comment:
The effects shouldn't be limited by CSound; in fact, the opposite will be true. CSound gives an extremely powerful way to create effects by combining filters and other primitive elements. The effects I will be designing will work at a higher level of abstraction, and will probably cover only a small subset of what CSound is capable of.

Compression and EQ are probably the two most important effects to implement; either one would be a glaring omission from a music production language. I would probably prioritize those two, and then time permitting add in others like delay, reverb, distortion and so on.